### ISSUE TITLE
Can't make this find issues

### ISSUE THREAD

**Issue by mgzenitech** [ota-meshi/eslint-plugin-json-schema-validator#328]
I have a file:
```
.vscode/settings.1.globals.ui.json
```
with content:
```
{
  "workbench.startupEditor": 1
}
```

and eslint config:
```
const eslintPluginJsonSchemaValidator = require('eslint-plugin-json-schema-validator');
...
module.exports = [
...
  ...eslintPluginJsonSchemaValidator.configs['flat/recommended'],
  {
    files: ["**/*.json", "**/*.json5", "**/*.jsonc"],
    rules: {
      "json-schema-validator/no-invalid": [
        "error",
        {
          schemas: [
            {
              fileMatch: ["./.vscode/settings.1.globals.ui.json"],
              schema: "https://raw.githubusercontent.com/wraith13/vscode-schemas/master/en/latest/schemas/settings/folder.json"
            }
          ],
          "useSchemastoreCatalog": false,
        }
      ]
    }
  }
];
```

Clearly based on the schema:
```
"workbench.startupEditor": 1
```

is incorrect value, however eslint doesn't detect anything with:
```
eslint -c .internal/configs/eslint.js .vscode/settings.1.globals.ui.json --debug
```
```
eslint:cli CLI args: [ '-c', '.internal/configs/eslint.js', '.vscode/settings.1.globals.ui.json', '--debug' ] +0ms
  eslint:cli Using flat config? true +2ms
  eslint:cli Running on files +3ms
  eslint:eslint Using file patterns: .vscode/settings.1.globals.ui.json +0ms
  eslint:eslint Override config file path is .internal/configs/eslint.js +0ms
  eslint:eslint Loading config from /home/xxx/Documents/Projects/yyy/zzz/.internal/configs/eslint.js +0ms
  eslint:eslint Config file URL is file:///home/xxx/Documents/Projects/yyy/zzz/.internal/configs/eslint.js +0ms
  eslint:rules Loading rule 'no-dupe-keys' (remaining=289) +0ms
  eslint:rules Loading rule 'no-floating-decimal' (remaining=288) +7ms
  eslint:rules Loading rule 'no-irregular-whitespace' (remaining=287) +4ms
  eslint:rules Loading rule 'no-multi-str' (remaining=286) +2ms
  eslint:rules Loading rule 'no-octal-escape' (remaining=285) +4ms
  eslint:rules Loading rule 'no-octal' (remaining=284) +3ms
  eslint:rules Loading rule 'no-sparse-arrays' (remaining=283) +3ms
  eslint:rules Loading rule 'no-useless-escape' (remaining=282) +4ms
  eslint:eslint Deleting cache file at /home/xxx/Documents/Projects/yyy/zzz/.eslintcache +476ms
  eslint:eslint 1 files found in: 41ms +41ms
  eslint:eslint Lint /home/xxx/Documents/Projects/yyy/zzz/.vscode/settings.1.globals.ui.json +2ms
  eslint:linter Linting code for /home/xxx/Documents/Projects/yyy/zzz/.vscode/settings.1.globals.ui.json (pass 1) +0ms
  eslint:linter Verify +0ms
  eslint:linter With flat config: /home/xxx/Documents/Projects/yyy/zzz/.vscode/settings.1.globals.ui.json +0ms
  eslint:linter Parsing: /home/xxx/Documents/Projects/yyy/zzz/.vscode/settings.1.globals.ui.json +0ms
  eslint:linter Parsing successful: /home/xxx/Documents/Projects/yyy/zzz/.vscode/settings.1.globals.ui.json +18ms
  eslint:linter Scope analysis: /home/xxx/Documents/Projects/yyy/zzz/.vscode/settings.1.globals.ui.json +0ms
  eslint:linter Scope analysis successful: /home/xxx/Documents/Projects/yyy/zzz/.vscode/settings.1.globals.ui.json +1ms
  eslint:linter Generating fixed text for /home/xxx/Documents/Projects/yyy/zzz/.vscode/settings.1.globals.ui.json (pass 1) +27ms
  eslint:source-code-fixer Applying fixes +0ms
  eslint:source-code-fixer shouldFix parameter was false, not attempting fixes +0ms
```

**Comment by mgzenitech** [comment]
just in case complete print of flag config:
```
{
  "linterOptions": {
    "reportUnusedDisableDirectives": 2
  },
  "languageOptions": {
    "ecmaVersion": "latest",
    "sourceType": "module",
    "parser": "jsonc-eslint-parser@2.4.0",
    "parserOptions": {},
    "globals": {}
  },
  "plugins": [
    "@",
    "jsonc:eslint-plugin-jsonc@2.15.1",
    "@stylistic/js",
    "json-schema-validator:eslint-plugin-json-schema-validator@5.1.0"
  ],
  "rules": {
    "strict": [
      0
    ],
    "no-unused-expressions": [
      0
    ],
    "no-unused-vars": [
      0
    ],
    "jsonc/array-bracket-newline": [
      2,
      {
        "multiline": true,
        "minItems": 1
      }
    ],
    "jsonc/array-bracket-spacing": [
      2
    ],
    "jsonc/array-element-newline": [
      2,
      {
        "minItems": 1
      }
    ],
    "jsonc/auto": [
      2
    ],
    "jsonc/comma-dangle": [
      2
    ],
    "jsonc/comma-style": [
      2
    ],
    "jsonc/indent": [
      2,
      2
    ],
    "jsonc/key-name-casing": [
      2,
      {
        "ignores": [
          "."
        ],
        "camelCase": true,
        "PascalCase": false,
        "SCREAMING_SNAKE_CASE": false,
        "kebab-case": false,
        "snake_case": false
      }
    ],
    "jsonc/key-spacing": [
      2
    ],
    "jsonc/no-bigint-literals": [
      2
    ],
    "jsonc/no-binary-expression": [
      2
    ],
    "jsonc/no-binary-numeric-literals": [
      2
    ],
    "jsonc/no-comments": [
      0
    ],
    "jsonc/no-dupe-keys": [
      2
    ],
    "jsonc/no-escape-sequence-in-identifier": [
      2
    ],
    "jsonc/no-floating-decimal": [
      2
    ],
    "jsonc/no-hexadecimal-numeric-literals": [
      2
    ],
    "jsonc/no-infinity": [
      2
    ],
    "jsonc/no-irregular-whitespace": [
      2
    ],
    "jsonc/no-multi-str": [
      2
    ],
    "jsonc/no-nan": [
      2
    ],
    "jsonc/no-number-props": [
      2
    ],
    "jsonc/no-numeric-separators": [
      2
    ],
    "jsonc/no-octal-escape": [
      2
    ],
    "jsonc/no-octal-numeric-literals": [
      2
    ],
    "jsonc/no-octal": [
      2
    ],
    "jsonc/no-parenthesized": [
      2
    ],
    "jsonc/no-plus-sign": [
      2
    ],
    "jsonc/no-regexp-literals": [
      2
    ],
    "jsonc/no-sparse-arrays": [
      2
    ],
    "jsonc/no-template-literals": [
      2
    ],
    "jsonc/no-undefined-value": [
      2
    ],
    "jsonc/no-unicode-codepoint-escapes": [
      2
    ],
    "jsonc/no-useless-escape": [
      2
    ],
    "jsonc/object-curly-newline": [
      2
    ],
    "jsonc/object-curly-spacing": [
      2
    ],
    "jsonc/object-property-newline": [
      2
    ],
    "jsonc/quote-props": [
      2
    ],
    "jsonc/quotes": [
      2
    ],
    "jsonc/sort-keys": [
      2,
      "asc",
      {
        "caseSensitive": false,
        "natural": true
      }
    ],
    "jsonc/space-unary-ops": [
      2
    ],
    "jsonc/valid-json-number": [
      2
    ],
    "jsonc/vue-custom-block/no-parsing-error": [
      2
    ],
    "jsonc/sort-array-values": [
      0
    ],
    "@stylistic/js/max-len": [
      2,
      {
        "code": 120
      }
    ],
    "@stylistic/js/no-multiple-empty-lines": [
      2,
      {
        "max": 0,
        "maxBOF": 0,
        "maxEOF": 1
      }
    ],
    "@stylistic/js/no-trailing-spaces": [
      2
    ],
    "@stylistic/js/eol-last": [
      2
    ],
    "@stylistic/js/spaced-comment": [
      2
    ],
    "@stylistic/js/linebreak-style": [
      2
    ],
    "json-schema-validator/no-invalid": [
      2,
      {
        "schemas": [
          {
            "fileMatch": [
              "./.vscode/settings.1.globals.ui.json"
            ],
            "schema": "https://raw.githubusercontent.com/wraith13/vscode-schemas/master/en/latest/schemas/settings/folder.json"
          }
        ],
        "useSchemastoreCatalog": false
      }
    ]
  }
}
```

**Comment by ota-meshi** [comment]
Could you please provide a GitHub repository to reproduce the issue?

**Comment by mgzenitech** [comment]
> Could you please provide a GitHub repository to reproduce the issue?

sure @ https://github.com/mgzenitech/schema-validation-repro-temp

**Comment by ota-meshi** [comment]
Thanks for sharing your repository.
It looks like you need to change the fileMatch.


```diff
-fileMatch: ["./.vscode/settings.1.globals.ui.json"],
+fileMatch: [".vscode/settings.1.globals.ui.json"],
```


But `https://raw.githubusercontent.com/wraith13/vscode-schemas/master/en/latest/schemas/settings/folder.json` doesn't seem to be a valid JSON schema. ajv throws an error when trying to parse it.

**Comment by mgzenitech** [comment]
> Thanks for sharing your repository. It looks like you need to change the fileMatch.
> 
> ```diff
> -fileMatch: ["./.vscode/settings.1.globals.ui.json"],
> +fileMatch: [".vscode/settings.1.globals.ui.json"],
> ```
> 
> But `https://raw.githubusercontent.com/wraith13/vscode-schemas/master/en/latest/schemas/settings/folder.json` doesn't seem to be a valid JSON schema. ajv throws an error when trying to parse it.

It's same schema that is being used internally by vscode when accessing via vscode protocol:
```
vscode://schemas/settings/folder
```

You can even set in .vscode/settings.json:
```
  "json.schemas": [
    {
      "fileMatch": [
        "/.vscode/settings.*.json"
      ],
      "url": "https://raw.githubusercontent.com/wraith13/vscode-schemas/master/en/latest/schemas/settings/folder.json"
    },
  }
```

**Comment by ota-meshi** [comment]
I think this is either an AJV or JSON Schema definition issue, so please contact one of the repositories.
